tests/unit/general/test_umpire_mem.cpp

```diff
@@ -162,7 +174,8 @@ static void test_umpire_device_memory()
    // Just as an example, temp memory on the stack is automatically cleaned up
    {
       printf("Allocate %u more bytes in temporary memory: ", num_bytes);
-      Vector dev_temp(num_elems, mc::DEVICE_TEMP);
+      Vector dev_temp(num_elems, MemoryType::DEVICE_UMPIRE_2);
+      dev_temp.Write(); // make sure device memory is allocated
```

Comment:
Isn't it true that when a device `MemoryType` is used the memory is only allocated on the device? Given that the `REQUIRE` below passes it seems like it is. Would this change?

Reply:
As mentioned in a previous comment, we always perform the host allocation. When the device allocation actually happens (with the constructor used here) is unspecified at the moment. I was thinking of specifying this to be always a "lazy" allocation for the device pointers (currently some device allocations are not "lazy", I think).